src/plugins/audioplayers/player.c

```diff
@@ -57,6 +66,31 @@ static int on_bus_fd_ready(sd_event_source *src, int fd, uint32_t revents, void
     return 0;
 }
 
+static void audio_player_source_setup(GstElement *playbin, GstElement *source, GstElement **p_src) {
+    (void)(playbin);
+    (void)(p_src);
+
+    if (g_object_class_find_property(G_OBJECT_GET_CLASS(source), "ssl-strict") != 0) {
+        g_object_set(G_OBJECT(source), "ssl-strict", FALSE, NULL);
+    }
+}
+
+static int on_receive_event_ch(char *channel, struct platch_obj *object, FlutterPlatformMessageResponseHandle *responsehandle) {
+    //TODO: Do we need any more setup to send events to dart code?
```

Comment:
This is a bit unclear to me as it seems flutter calls these methods to notify when to listen or cancel.
So I assume I might need to only send event if someone listens or maybe I could just send all the time?

Reply:
Yes, you should only send events when someone is listening on the channel, i.e. after `listen` was called and before `cancel` is called.

If you send any events when nothing is listening on the channel, it will be buffered on the dart side, and once the dart side actually starts listening on the channel, this buffer will be drained and the channel handler will receive the buffered messages. Though the buffer has a capacity of only 1 message by default, so if you emit more than one message while nothing's listening, it will print a warning: https://github.com/flutter/engine/blob/875d87e02276c17d8224953bf72e1a93b07a5679/lib/ui/channel_buffers.dart#L332-L340

Reply:
Thanks for clarification.
I should keep track on number of listeners then (although in practice `audioplayers` immediately listens on player creation)`

Reply:
There can only be one listener at a time, so you should just have a bool `has_listener` or sth

Reply:
oh I see, ok then.
I will implement it and test again on weekend

Reply:
@ardera This is fixed now. I tested plugin to make sure it register channel successfully too so it looks good